platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java

```diff
@@ -1046,9 +1051,13 @@ private void replayPreconsensusEvents() {
                     consensusRoundHandler,
                     stateHashSignQueue,
                     stateManagementComponent,
-                    currentPlatformStatus::get,
-                    this::setPlatformStatus,
+                    platformStatusManager,
                     initialMinimumGenerationNonAncient);
+        } else {
+            // if preconsensus events aren't being replayed, advance through that part of the state machine anyway
+            platformStatusManager.submitStatusAction(new StartedReplayingEventsAction());
+            platformStatusManager.submitStatusAction(
+                    new DoneReplayingEventsAction(Time.getCurrent().now()));
```

Comment:
If you pull the status action stuff out of `PreconsensusEventReplayWorkflow.replayPreconsensusEvents()`, then you can delete this else block and rearrange the code like this:

```
        platformStatusManager.submitStatusAction(new StartedReplayingEventsAction());
        if (enableReplay) {
            PreconsensusEventReplayWorkflow.replayPreconsensusEvents(
                    platformContext,
                    threadManager,
                    Time.getCurrent(),
                    preconsensusEventFileManager,
                    preconsensusEventWriter,
                    eventTaskDispatcher,
                    intakeQueue,
                    consensusRoundHandler,
                    stateHashSignQueue,
                    stateManagementComponent,
                    currentPlatformStatus::get,
                    this::setPlatformStatus,
                    platformStatusManager,
                    initialMinimumGenerationNonAncient);
        }
        platformStatusManager.submitStatusAction(
                    new DoneReplayingEventsAction(Time.getCurrent().now()));
```

Reply:
A thought, feel free to take it or leave it depending if you think it is more or less elegant.

What if we hid the action objects from the public API, e.g. `new DoneReplayingEventsAction(Time.getCurrent().now())`. Instead, there would be a method for each action type, e.g. `platformStatusManager.doneReplayingEvents() // time is captured internally`.  We'd still use the action objects internally on the queue, but the use of the status manager becomes simpler.

If you end up doing this, the interface `StatusActionSubmitter` would end up having these methods on it, and `PlatformStatusStateMachine` would implement those methods.

If you like this idea but think it's too big a change at this point in the PR's lifespan, feel free to instead do it as a follow up task.

Reply:
I like the idea of hiding the actions - I created an issue for this work [here](https://github.com/hashgraph/hedera-services/issues/7618)